public/pages/Findings/containers/Findings/Findings.tsx

```diff
@@ -157,44 +157,57 @@ class Findings extends Component<FindingsProps, FindingsState> {
     const { findingsService, detectorService, notifications } = this.props;
     try {
-      const detectorsRes = await detectorService.getDetectors();
-      if (detectorsRes.ok) {
-        const detectors = detectorsRes.response.hits.hits;
+      // Get all the Findings logic
+      const findingRes = await findingsService.getFindings();
+
+      if (findingRes.ok) {
         const ruleIds = new Set<string>();
         let findings: FindingItemType[] = [];
 
-        const detectorId = this.props.match.params['detectorId'];
-        for (let detector of detectors) {
-          if (!detectorId || detector._id === detectorId) {
-            const findingRes = await findingsService.getFindings({ detectorId: detector._id });
-
-            if (findingRes.ok) {
-              const detectorFindings: FindingItemType[] = findingRes.response.findings.map(
-                (finding) => {
-                  finding.queries.forEach((rule) => ruleIds.add(rule.id));
-                  return {
-                    ...finding,
-                    detectorName: detector._source.name,
-                    logType: detector._source.detector_type,
-                    detector: detector,
-                  };
-                }
-              );
-              findings = findings.concat(detectorFindings);
-            } else {
-              errorNotificationToast(notifications, 'retrieve', 'findings', findingRes.error);
+        // Assuming getDetectors returns an array of detectors
+        const detectorsRes = await detectorService.getDetectors();
+        if (detectorsRes.ok) {
+          const detectors = detectorsRes.response.hits.hits;
+          const detectorInfoMap = new Map();
+
+          detectors.forEach((detector) => {
+            detectorInfoMap.set(detector._id, {
+              logType: detector._source.detector_type,
+              name: detector._source.name,
+              detector: detector,
+            });
+          });
+          const detectorFindingsPromises = findingRes.response.findings.map(async (finding) => {
+            try {
+              // Use detectorInfoMap to get detector information
+              const detectorInfo = detectorInfoMap.get(finding.detectorId);
+              finding.queries.forEach((rule) => ruleIds.add(rule.id));
+              return {
+                ...finding,
+                detectorName: detectorInfo ? detectorInfo.name : 'Unknown',
+                logType: detectorInfo ? detectorInfo.logType : 'Unknown',
+                detector: detectorInfo.detector, // If needed, replace with actual detector information
+              };
+            } catch (error) {
+              console.error('Error while fetching detector information:', error);
+              errorNotificationToast(notifications, 'retrieve', 'findings', error);
+              return null;
             }
-          }
+          });
+
+          const detectorFindings = await Promise.all(detectorFindingsPromises);
```

Comment:
Similar to the question I had on line 180, are we sure these are promises? I believe we're already `await`ing the response from the API calls on lines 161 and 168, but I may have missed something.

Reply:
Yes, the await keyword is required in this context. The Promise.all function returns a single promise that resolves to an array of the fulfilled values of the input promises (or rejects with the reason of the first promise that rejects, if any). In order to wait for all the promises inside detectorFindingsPromises to resolve before moving on to the next line of code, we are using await with Promise.all.